server/deploys_resource.go

```diff
@@ -152,28 +142,35 @@ func (dr *DeploysResource) destroyPrevious(name string, previousVersion string,
 		timeoutChan <- true
 	}()
 
+	currentFleetUnit := fleetServiceName(name, currentVersion)
+	previousFleetUnit := fleetServiceName(name, previousVersion)
+
 	for {
-		startCheck := time.After(destroyPreviousCheckDelay)
+		startCheck := time.After(checkDelay)
 
 		select {
 		case <-startCheck:
-			log.Printf("Checking if %s:%s has finished launching...\n", name, currentVersion)
-			state, err := dr.Fleet.UnitState(fleetServiceName(name, currentVersion))
```

Comment:
The official Fleet client doesn't support returning only states that match a given unit name.  This means we have to iterate through all the results.  Its possible we could contribute something here to add this ability upstream given that its in the API documentation for Fleet.